eth/vm/rstack.py

```diff
@@ -0,0 +1,68 @@
+from typing import (
+    List,
+    Tuple,
+    Union,
+)
+
+from eth.exceptions import (
+    InsufficientStack,
+    FullStack,
+)
+
+from eth.validation import (
+    validate_stack_int,
+)
+
+from eth_utils import (
+    big_endian_to_int,
+    ValidationError,
+)
+
+"""
+This module simply implements for the return stack the exact same design used for the data stack.
+As this stack must simply push_int or pop1_int any time a subroutine is accessed or left, only
+those two functions are provided.
+For the same reason, the class RStack doesn't inherit from the abc StackAPI, as it would require
+to implement all the abstract methods defined.
+"""
+
+
+class RStack:
+    """
+    VM Return Stack
+    """
+
+    __slots__ = ['values', '_append', '_pop_typed', '__len__']
+
+    def __init__(self) -> None:
+        values: List[Tuple[type, Union[int, bytes]]] = []
+        self.values = values
+        self._append = values.append
+        self._pop_typed = values.pop
+        self.__len__ = values.__len__
+
+    def push_int(self, value: int) -> None:
```

Comment:
Since there aren't two type options anymore, this method can just be a push.
```suggestion
    def push(self, value: int) -> None:
```